src/client/apolloClient.ts

```diff
@@ -80,5 +86,20 @@ export const apolloClient = (() => {
     ]),
   })
 
-  return { subscriptionClient, client, httpLink }
+  const runQuery = async <T = any, TVariables = OperationVariables>(
+    options: QueryOptions<TVariables, T>,
+  ): Promise<T | undefined> => {
+    try {
+      const apolloQuery = await client.query<T>(options)
+      if (apolloQuery.error) {
+        datadogRum.addError(apolloQuery.error, options)
+        return
+      } else return apolloQuery.data
+    } catch (e) {
+      datadogRum.addError(e, options)
+      return
+    }
```

Comment:
Do we need to send any more context to datadog here? Or maybe it's gonna get picked up automatically

Reply:
we pass the whole query and variables as context, if we'd need to pass more stuff explicitly we'd need it to supply it as a function argument from outside